web_programming/recaptcha_verification.py

```
@@ -0,0 +1,89 @@
+        <div class="g-recaptcha" data-sitekey="ClientKey"></div>
+      
+               
+    </form>
+
+    <!-- Below is the recaptcha script to be kept inside html tag -->
+    <script src="https://www.google.com/recaptcha/api.js" async defer></script>
+
+"""
+
+"""
+Below one Django function based code for views.py file for a login form has been shown with recaptcha verification
+"""
+
+
+def login_using_recaptcha(request):
+
+    # When Submit button is clicked
+    if request.method == "POST":
+        # get username, password & client_key from frontend
+        username = request.POST.get("username")
+        password = request.POST.get("password")
+        client_key = request.POST.get("g-recaptcha-response")
+
+        # Keep your recaptcha secret key here
+        secret_key = "secretKey"
+
+        # make json of your captcha data
+        captchaData = {"secret": secret_key, "response": client_key}
+
+        # post recaptcha response to Google recaptcha api
+        post = requests.post(
+            "https://www.google.com/recaptcha/api/siteverify", data=captchaData
+        )
+
+        # read the json response from recaptcha api
+        response = json.loads(post.text)
+        verify = response["success"]
+
+        # if verify is true
+        if verify == True:
+            # authenticate user
+            user = authenticate(request, username=username, password=password)
+
+            # if user is in database
+            if user is not None:
+                # login user
+                login(request, user)
+                return redirect("/your-webpage")
+            else:
+                # else send user back to the login page again
+                return render(request, "login.html")
+        else:
+            # if verify is not true, send user back to login page
+            return render(request, "login.html")
+
+    # return the login page when loading or when submit is not pressed direct user to login page
```

This is fail slow.  My suggestion was to fail fast.